R/zchunk_LA124.heat.R

```diff
@@ -40,70 +40,175 @@ module_energy_LA124.heat_DISABLED <- function(command, ...) {
     L1231.out_EJ_R_elec_F_tech_Yh <- get_data(all_data, "temp-data-inject/L1231.out_EJ_R_elec_F_tech_Yh")
 
     # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
+    # Create list of heat regions
+    heat_regionIDs <- A_regions$GCAM_region_ID[ A_regions$heat == 1]
+
+    # Fuel inputs to district heat
+    # Process fuel inputs in all regions; some will have the energy assigned to bld/ind, and others have a district heat sector
+    L1011.en_bal_EJ_R_Si_Fi_Yh  %>%
+      filter(sector == "in_heat") %>%
+      mutate(sector = sub("in_", "", sector)) %>%
+      left_join(enduse_fuel_aggregation, by="fuel") %>%
+      select(-fuel, -electricity, -bld, -industry, -trn) %>%
+      rename(fuel=heat) %>%
+      gather(key=year, value=value, -fuel, -sector, -GCAM_region_ID) %>%
+      group_by(fuel, sector, GCAM_region_ID, year) %>%
+      summarise(value=sum(value)) %>%
+      mutate(year = sub("X", "", year)) %>%
+      mutate(year = as.integer(year)) -> L124.in_EJ_R_heat_F_Yh
+
+    # Heat production from district heat sector
+    # Create empty table of historical years to fill with interpolation
+    A24.globaltech_coef[,as.character(HISTORICAL_YEARS[-1])] <- NA
```

Comment:
This is...weird. Is the `A24.globaltech_coef` frame wide? Why are you dropping the first column?